TestVectorGenerator/Complex.cs

```diff
@@ -28,6 +31,45 @@ public void Generate(Stream outputStream)
 			document.ContentKeys.Add(TestHelpers.GenerateContentKey());
 			document.ContentKeys.Add(TestHelpers.GenerateContentKey());
 
+			// Adding a non-random key, so its ID would match the key ID in the DRM
+			// system signaling data fields.
+			var contentKeyWithKnownId = new ContentKey
+			{
+				Id = Guid.Parse("f8c80c25-690f-4736-8132-430e5c6994ce"),
+				Value = new byte[] { 6, 7, 8, 9, 10, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5 }
+			};
+			document.ContentKeys.Add(contentKeyWithKnownId);
+
+			document.DrmSystems.Add(new DrmSystem
+			{
+				SystemId = Guid.Parse("edef8ba9-79d6-4ace-a3c8-27dcd51d21ed"),
+				KeyId = contentKeyWithKnownId.Id,
+				ContentProtectionData = "<cenc:pssh xmlns:cenc=\"urn:mpeg:cenc:2013\">AAAANHBzc2gAAAAA7e+LqXnWSs6jyCfc1R0h7QAAABQIARIQ+MgMJWkPRzaBMkMOXGmUzg==</cenc:pssh>",
+				HlsSignalingData = new HlsSignalingData
+				{
+					MasterPlaylistData = "#EXT-X-SESSION-KEY:METHOD=SAMPLE-AES,URI=\"data:text/plain;base64,AAAANHBzc2gAAAAA7e+LqXnWSs6jyCfc1R0h7QAAABQIARIQ+MgMJWkPRzaBMkMOXGmUzg==\",KEYID=0xF8C80C25690F47368132430E5C6994CE,KEYFORMAT=\"urn:uuid:edef8ba9-79d6-4ace-a3c8-27dcd51d21ed\",KEYFORMATVERSIONS=\"1\"",
+					VariantPlaylistData = "#EXT-X-KEY:METHOD=SAMPLE-AES,URI=\"data:text/plain;base64,AAAANHBzc2gAAAAA7e+LqXnWSs6jyCfc1R0h7QAAABQIARIQ+MgMJWkPRzaBMkMOXGmUzg==\",KEYID=0xF8C80C25690F47368132430E5C6994CE,KEYFORMAT=\"urn:uuid:edef8ba9-79d6-4ace-a3c8-27dcd51d21ed\",KEYFORMATVERSIONS=\"1\""
+				}
+			});
+			document.DrmSystems.Add(new DrmSystem
+			{
+				SystemId = Guid.Parse("9a04f079-9840-4286-ab92-e65be0885f95"),
+				KeyId = contentKeyWithKnownId.Id,
+				ContentProtectionData =
+					"<cenc:pssh xmlns:cenc=\"urn:mpeg:cenc:2013\">AAAB5HBzc2gAAAAAmgTweZhAQoarkuZb4IhflQAAAcTEAQAAAQABALoBPABXAFIATQBIAEUAQQBEAEUAUgAgAHgAbQBsAG4AcwA9ACIAaAB0AHQAcAA6AC8ALwBzAGMAaABlAG0AYQBzAC4AbQBpAGMAcgBvAHMAbwBmAHQALgBjAG8AbQAvAEQAUgBNAC8AMgAwADAANwAvADAAMwAvAFAAbABhAHkAUgBlAGEAZAB5AEgAZQBhAGQAZQByACIAIAB2AGUAcgBzAGkAbwBuAD0AIgA0AC4AMAAuADAALgAwACIAPgA8AEQAQQBUAEEAPgA8AFAAUgBPAFQARQBDAFQASQBOAEYATwA+ADwASwBFAFkATABFAE4APgAxADYAPAAvAEsARQBZAEwARQBOAD4APABBAEwARwBJAEQAPgBBAEUAUwBDAFQAUgA8AC8AQQBMAEcASQBEAD4APAAvAFAAUgBPAFQARQBDAFQASQBOAEYATwA+ADwASwBJAEQAPgBKAFEAegBJACsAQQA5AHAATgBrAGUAQgBNAGsATQBPAFgARwBtAFUAegBnAD0APQA8AC8ASwBJAEQAPgA8AC8ARABBAFQAQQA+ADwALwBXAFIATQBIAEUAQQBEAEUAUgA+AA==</cenc:pssh>" +
```

Comment:
Is all of this PSSH data valid? I remember you reported some bugs about it.

I recommend you make this all non-hardcoded valid data, generated by `MakePlayReadyPssh(KID)` methods and equivalent. You can just copypaste the PSSH creation methods from Makemedia - they already exist there in pretty much exactly the form you need here, I think. At least for HLS and DASH. I can later add Smooth Streaming and HDS valid sample data based on that if you do not know how they format the stuff.

Reply:
Should be all valid. But, I'll try to add generation code.